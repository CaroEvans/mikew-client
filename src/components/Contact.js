import React from 'react'
import TextField from 'material-ui/TextField'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'

class Contact extends React.Component{
    state= {
        fname: '',
        fnameError: '',
        lname: '',
        lnameError: '',
        phone: '',
        phoneError: '',
        email: '',
        emailError: '',
        artist: '',
        comment: ''
    }

    validate = () => {
        const { email, fname, lname, phone } = this.state
        let isError = false
        const errors = {
            fnameError: '',
            lnameError: '',
            phoneError: '',
            emailError: '',
        }

        if(fname.length <= 1 && isNaN(fname)){
            isError = true
            errors.fnameError = "please enter a valid first name."
        }

        if(lname.length <= 1 && isNaN(lname)){
            isError = true
            errors.lnameError = "please enter a valid last name."
        }

        if(email.indexOf("@") === -1 && email.indexOf(".") === -1 ){
            isError = true
            errors.emailError = "please enter a valid email address"
        }

        if(isNaN(parseInt(phone)) && phone.length < 5){
            isError = true
            errors.phoneError = "please enter a valid phone number"
        }

        this.setState ({
            ...this.state,
            ...errors
        })

        return isError
    }

    contactRequest = (e) => {
        e.preventDefault()

        const err = this.validate()
        console.log(
            this.state
        )

        //clear form
        if(!err) {
            this.setState({
                fname: '',
                fnameError: '',
                lname: '',
                lnameError: '',
                phone: '',
                phoneError: '',
                email: '',
                emailError: '',
                artist: '',
                comment: ''
            })
        }
    }

    handleChange = (e) => {
        this.setState({ [e.target.name]: e.target.value })
    }

    render(){
        const {fname, lname, phone, email, artist, comment, fnameError, lnameError, phoneError, emailError } = this.state
        const {handleChange} = this
        return(
            <MuiThemeProvider>
        <div className = "contact">
            <p> Contact Mike  </p>
            <form>
                <TextField
                    name= "fname" 
                    floatingLabelText= "first name"
                    value= {fname} 
                    onChange= {handleChange}
                    errorText = {fnameError} />
                <TextField
                    name= "lname" 
                    floatingLabelText= "last name"
                    value= {lname} 
                    onChange= {handleChange}
                    errorText = {lnameError}  />
                <TextField
                    name= "phone" 
                    floatingLabelText= "phone"
                    value= {phone} 
                    onChange= {handleChange}
                    errorText = {phoneError}  />
                <TextField
                    name= "email" 
                    floatingLabelText= "email"
                    value= {email} 
                    onChange= {handleChange}
                    errorText = {emailError}  />
                <TextField 
                    name= "artist" 
                    floatingLabelText= "artist"
                    value= {artist} 
                    onChange= {handleChange} />
                <TextField 
                    name= "comment" 
                    hintText= "What would you like to talk about?" 
                    floatingLabelText= "comment"
                    value= {comment} 
                    onChange= {handleChange} />
                <button onClick={this.contactRequest}> Send a Request </button>
            </form>
        </div>
        </MuiThemeProvider>
        )
    }
}
 
export default Contact;