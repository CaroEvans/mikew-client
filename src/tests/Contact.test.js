import React from 'react'
import Contact from '../components/Contact.js'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import './setUpTests'

describe('Contacts Component', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<Contact />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('sent state is false', () => {
        const wrapper = shallow(<Contact />)
        const sentState = wrapper.state().sent
        expect(sentState).toEqual("Send")
    })
    it('send button', () => {
      const wrapper = shallow(<Contact />)
      const text = wrapper.find('.sendContact').text()
      expect(text).toContain('Send')
  })
})