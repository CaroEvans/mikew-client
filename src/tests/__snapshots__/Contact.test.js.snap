// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts Component matches the snapshot 1`] = `
<div
  className="sc-htpNat kTRfOx"
>
  <div>
    <h4
      className="sc-bdVaJa esfNvT"
    >
       Get in Contact Easily 
    </h4>
    <h2
      className="sc-bwzfXH ePPsnT"
    >
        with Michael Waye 
    </h2>
    <div
      className="contact--form--container"
    >
      <form
        className="contact-form"
      >
        <p>
          <div
            className={undefined}
            style={
              Object {
                "backgroundColor": "transparent",
                "cursor": "auto",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 16,
                "height": 72,
                "lineHeight": "24px",
                "muiPrepared": true,
                "position": "relative",
                "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": 256,
              }
            }
          >
            <label
              className={undefined}
              htmlFor="fname-undefined-firstname-21303"
              onClick={undefined}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.3)",
                  "lineHeight": "22px",
                  "muiPrepared": true,
                  "pointerEvents": "none",
                  "position": "absolute",
                  "top": 38,
                  "transform": "scale(1) translate(0, 0)",
                  "transformOrigin": "left top",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "zIndex": 1,
                }
              }
            >
              first name
            </label>
            <input
              disabled={false}
              id="fname-undefined-firstname-21303"
              name="fname"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "WebkitOpacity": 1,
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "appearance": "textfield",
                  "backgroundColor": "rgba(0,0,0,0)",
                  "border": "none",
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "cursor": "inherit",
                  "font": "inherit",
                  "height": "100%",
                  "marginTop": 14,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "width": "100%",
                }
              }
              type="text"
              value=""
            />
            <div>
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              />
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 2,
                    "borderColor": "#00bcd4",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "transform": "scaleX(0)",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": "100%",
                  }
                }
              />
            </div>
            
          </div>
        </p>
        <p>
          <div
            className={undefined}
            style={
              Object {
                "backgroundColor": "transparent",
                "cursor": "auto",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 16,
                "height": 72,
                "lineHeight": "24px",
                "muiPrepared": true,
                "position": "relative",
                "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": 256,
              }
            }
          >
            <label
              className={undefined}
              htmlFor="lname-undefined-lastname-55270"
              onClick={undefined}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.3)",
                  "lineHeight": "22px",
                  "muiPrepared": true,
                  "pointerEvents": "none",
                  "position": "absolute",
                  "top": 38,
                  "transform": "scale(1) translate(0, 0)",
                  "transformOrigin": "left top",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "zIndex": 1,
                }
              }
            >
              last name
            </label>
            <input
              disabled={false}
              id="lname-undefined-lastname-55270"
              name="lname"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "WebkitOpacity": 1,
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "appearance": "textfield",
                  "backgroundColor": "rgba(0,0,0,0)",
                  "border": "none",
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "cursor": "inherit",
                  "font": "inherit",
                  "height": "100%",
                  "marginTop": 14,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "width": "100%",
                }
              }
              type="text"
              value=""
            />
            <div>
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              />
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 2,
                    "borderColor": "#00bcd4",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "transform": "scaleX(0)",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": "100%",
                  }
                }
              />
            </div>
            
          </div>
        </p>
        <p>
          <div
            className={undefined}
            style={
              Object {
                "backgroundColor": "transparent",
                "cursor": "auto",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 16,
                "height": 72,
                "lineHeight": "24px",
                "muiPrepared": true,
                "position": "relative",
                "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": 256,
              }
            }
          >
            <label
              className={undefined}
              htmlFor="phone-undefined-phone-45484"
              onClick={undefined}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.3)",
                  "lineHeight": "22px",
                  "muiPrepared": true,
                  "pointerEvents": "none",
                  "position": "absolute",
                  "top": 38,
                  "transform": "scale(1) translate(0, 0)",
                  "transformOrigin": "left top",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "zIndex": 1,
                }
              }
            >
              phone
            </label>
            <input
              disabled={false}
              id="phone-undefined-phone-45484"
              name="phone"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "WebkitOpacity": 1,
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "appearance": "textfield",
                  "backgroundColor": "rgba(0,0,0,0)",
                  "border": "none",
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "cursor": "inherit",
                  "font": "inherit",
                  "height": "100%",
                  "marginTop": 14,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "width": "100%",
                }
              }
              type="text"
              value=""
            />
            <div>
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              />
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 2,
                    "borderColor": "#00bcd4",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "transform": "scaleX(0)",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": "100%",
                  }
                }
              />
            </div>
            
          </div>
        </p>
        <p>
          <div
            className={undefined}
            style={
              Object {
                "backgroundColor": "transparent",
                "cursor": "auto",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 16,
                "height": 72,
                "lineHeight": "24px",
                "muiPrepared": true,
                "position": "relative",
                "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": 256,
              }
            }
          >
            <label
              className={undefined}
              htmlFor="email-undefined-email-3074"
              onClick={undefined}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.3)",
                  "lineHeight": "22px",
                  "muiPrepared": true,
                  "pointerEvents": "none",
                  "position": "absolute",
                  "top": 38,
                  "transform": "scale(1) translate(0, 0)",
                  "transformOrigin": "left top",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "zIndex": 1,
                }
              }
            >
              email
            </label>
            <input
              disabled={false}
              id="email-undefined-email-3074"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "WebkitOpacity": 1,
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "appearance": "textfield",
                  "backgroundColor": "rgba(0,0,0,0)",
                  "border": "none",
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "cursor": "inherit",
                  "font": "inherit",
                  "height": "100%",
                  "marginTop": 14,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "width": "100%",
                }
              }
              type="text"
              value=""
            />
            <div>
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              />
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 2,
                    "borderColor": "#00bcd4",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "transform": "scaleX(0)",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": "100%",
                  }
                }
              />
            </div>
            
          </div>
        </p>
        <p>
          <div
            className={undefined}
            style={
              Object {
                "backgroundColor": "transparent",
                "cursor": "auto",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 16,
                "height": 72,
                "lineHeight": "24px",
                "muiPrepared": true,
                "position": "relative",
                "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": 256,
              }
            }
          >
            <label
              className={undefined}
              htmlFor="artist-undefined-artist-47426"
              onClick={undefined}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.3)",
                  "lineHeight": "22px",
                  "muiPrepared": true,
                  "pointerEvents": "none",
                  "position": "absolute",
                  "top": 38,
                  "transform": "scale(1) translate(0, 0)",
                  "transformOrigin": "left top",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "zIndex": 1,
                }
              }
            >
              artist
            </label>
            <input
              disabled={false}
              id="artist-undefined-artist-47426"
              name="artist"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "WebkitOpacity": 1,
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "appearance": "textfield",
                  "backgroundColor": "rgba(0,0,0,0)",
                  "border": "none",
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "cursor": "inherit",
                  "font": "inherit",
                  "height": "100%",
                  "marginTop": 14,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "width": "100%",
                }
              }
              type="text"
              value=""
            />
            <div>
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              />
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 2,
                    "borderColor": "#00bcd4",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "transform": "scaleX(0)",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
        </p>
        <p>
          <div
            className={undefined}
            style={
              Object {
                "backgroundColor": "transparent",
                "cursor": "auto",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 16,
                "height": 72,
                "lineHeight": "24px",
                "muiPrepared": true,
                "position": "relative",
                "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": 256,
              }
            }
          >
            <label
              className={undefined}
              htmlFor="comment-Whatwouldyouliketotalkabout-comment-13712"
              onClick={undefined}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.3)",
                  "lineHeight": "22px",
                  "muiPrepared": true,
                  "pointerEvents": "none",
                  "position": "absolute",
                  "top": 38,
                  "transform": "scale(1) translate(0, 0)",
                  "transformOrigin": "left top",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "zIndex": 1,
                }
              }
            >
              comment
            </label>
            <div
              style={
                Object {
                  "bottom": 12,
                  "color": "rgba(0, 0, 0, 0.3)",
                  "muiPrepared": true,
                  "opacity": 0,
                  "position": "absolute",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            >
              What would you like to talk about?
            </div>
            <input
              disabled={false}
              id="comment-Whatwouldyouliketotalkabout-comment-13712"
              name="comment"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "WebkitOpacity": 1,
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "appearance": "textfield",
                  "backgroundColor": "rgba(0,0,0,0)",
                  "border": "none",
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "cursor": "inherit",
                  "font": "inherit",
                  "height": "100%",
                  "marginTop": 14,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "width": "100%",
                }
              }
              type="text"
              value=""
            />
            <div>
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              />
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 2,
                    "borderColor": "#00bcd4",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "transform": "scaleX(0)",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
        </p>
      </form>
    </div>
    <button
      className="sendContact"
      disabled={false}
      onClick={[Function]}
    >
       
      Send
       
    </button>
     
  </div>
</div>
`;
